src/renderer/api/helpers/watch-for-general-entity-navigation.injectable.ts

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectable } from "@ogre-tools/injectable";
+import { reaction, when } from "mobx";
+import generalCategoryInjectable from "../../../common/catalog/categories/general.injectable";
+import isActiveRouteInjectable from "../../navigation/is-route-active.injectable";
+import observableHistoryInjectable from "../../navigation/observable-history.injectable";
+import type { Disposer } from "../../utils";
+import { disposer } from "../../utils";
+import catalogEntityRegistryInjectable from "../catalog-entity-registry/catalog-entity-registry.injectable";
+
+export type WatchForGeneralEntityNavigation = () => Disposer;
+
+const watchForGeneralEntityNavigationInjectable = getInjectable({
+  id: "watch-for-general-entity-navigation",
+  instantiate: (di): WatchForGeneralEntityNavigation => {
+    const observableHistory = di.inject(observableHistoryInjectable);
+    const isActiveRoute = di.inject(isActiveRouteInjectable);
+    const entityRegistry = di.inject(catalogEntityRegistryInjectable);
+    const generalCategory = di.inject(generalCategoryInjectable);
+
+    return () => {
+      const dispose = disposer();
+
+      dispose.push(when(
+        () => entityRegistry.entities.size > 0,
+        () => {
+          dispose.push(reaction(
+            () => observableHistory.location,
+            () => {
+              const entities = entityRegistry.getItemsForCategory(generalCategory);
+              const activeEntity = entities.find(entity => isActiveRoute(entity.spec.path));
+
+              if (activeEntity) {
+                entityRegistry.activeEntity = activeEntity;
+              }
+            },
+            {
+              fireImmediately: true,
+            },
+          ));
+        },
+      ));
```

Comment:
Untolerable complexity? I'd say so.

Reply:
Can you suggest a better alternative then?

Reply:
1. Rename so that "watch-for-general-entity-navigation" instead explains the motivation of watching.
2. Create reactive dependencies (i.e. `IComputedValue`) instead of this statefulness. E.g. could this just be `activeEntityInjectable`?
3. Consider `when` and `reaction` as reasons to suspect something might be off because of statefulness and non-reactiveness.
4. Add functions and variables with names explaining motivation.
5. Consider deeply nested code as questionable.
6. This is a mutation of an observable from a reacted observation in other observables. I thought MobX enforced prevention of this.
7. Consider if, instead of reaction, this functionality can be relocated to the action causing the originating observation somehow.

Reply:
You wrote this code. I wrote this code. Our team wrote this code. It is our code :) 

For me, this code is hairy enough to be  fixed right away, for the reputation of the team. But I respect your decision to see it differently. 

Note: PR's or process is not a good excuse to not fix this. It's just a commit.

Resolve on will.

Reply:
Yes I think that `activeEntity` should be completely reactive, but that causes a whole host of other issues because of the `iframes`.

Will resolve for the time being.